### ISSUE TITLE
Crash Report: Cannot read properties of undefined (reading 'level')

### ISSUE THREAD

**Issue by SporyTike** [pagefaultgames/pokerogue#1415]
Currently I am in the middle of a classic run and I can't continue it due to a crash. Everytime I reach floor 91 in my current run, the game crashes and throws the following error into the console.
![image](https://github.com/pagefaultgames/pokerogue/assets/45672834/9946d984-4392-43ca-a96f-4c29d37245e5)

From the error I assume it has something to do with what the game tries to load next, but I am not totally sure so I want to give as much information as possible.
I am in a classic mode run on floor 90 which has a Machamp boss. When I defeated it the game continues up to the point where the trainer would normally throw out the Pokémon for the next biom. I tried loading the game several time in floor 90 to try defeating Machamp on different ways to get a different RNG, but it seems to not work. I also used the older backup from floor 83 with the hope that I can bypass the crash that way but without any luck  and I have no idea if there is something else I could try to save the run. I am not totally sure what else I can provide but for now I keep the run on one of my slots in case it is needed and can be extracted somehow.

**Comment by tripleawwy** [comment]
this is where it breaks for me:
![image](https://github.com/pagefaultgames/pokerogue/assets/43851917/896b55b0-8736-4cfd-8df2-779c643fa0c8)

with the exact same error message in the console
```
index-7b9ac88b.js:12329 TypeError: Cannot read properties of undefined (reading 'level')
    at OF.getSimulatedEvolutionChain (index-7b9ac88b.js:11131:10712)
    at jk.getLevelMoves (index-7b9ac88b.js:11645:8886)
    at jk.generateAndPopulateMoveset (index-7b9ac88b.js:11645:11714)
    at jk.generateAndPopulateMoveset (index-7b9ac88b.js:11645:47392)
    at new EnemyPokemon (index-7b9ac88b.js:11645:46101)
    at R2.addEnemyPokemon (index-7b9ac88b.js:12197:30291)
    at Object.-2 (index-7b9ac88b.js:11502:10811)
    at index-7b9ac88b.js:11502:74251
    at R2.executeWithSeedOffset (index-7b9ac88b.js:12197:39066)
    at VF.genPartyMember (index-7b9ac88b.js:11502:73888)
```

ps: it was a machoke for me

**Comment by Madmadness65** [comment]
Should be fixed by [72c7e9c](https://github.com/pagefaultgames/pokerogue/commit/72c7e9c194d453bd6ac9d1ad567463bc6ae6a93c) now
